auth/simple_token.go

```diff
@@ -52,6 +54,21 @@ func (as *authStore) assignSimpleTokenToUser(username, token string) {
 	}
 
 	as.simpleTokens[token] = username
+	deleteTokenFunc := func() {
+		as.simpleTokensMu.Lock()
+		_, ok := as.simpleTokenLock[token]
+		if ok {
+			delete(as.simpleTokenLock, token)
+			as.simpleTokensMu.Unlock()
+			return
+		}
+		delete(as.simpleTokens, token)
+		delete(as.simpleTokenInf, token)
+		as.simpleTokensMu.Unlock()
+		plog.Errorf("deleted the token %s for user %s", token, username)
+	}
+	timer := time.AfterFunc(SimpleTokenTTL, deleteTokenFunc)
```

Comment:
the race condition that timer is fired but not executed is also handled using the flag being set in map simpleTokenLock and reset in the closure function given to time.AfterFunc